lib/runtime/src/vm.rs

```diff
@@ -28,12 +29,51 @@ pub struct Ctx {
 
     /// The parent instance.
     pub local_backing: *mut LocalBacking,
+
+    /// Host data
+    pub data: *mut c_void,
+
+    /// Host data finalizer
+    pub data_finalizer: extern "C" fn(data: *mut c_void),
+
+}
+
+extern "C" fn empty_finalizer(_data: *mut c_void){
+
+}
+
+impl Drop for Ctx {
+    fn drop(&mut self) {
+        (self.data_finalizer)(self.data);
+    }
 }
 
 impl Ctx {
     pub unsafe fn new(
+        local_backing: &mut LocalBacking,
+        import_backing: &mut ImportBacking
+    ) -> Self {
```

Comment:
It might be a good idea to reuse `new_with_data`.

```rust
Self::new_with_data(local_backing, import_backing, ptr::null_mut(), empty_finalizer)
```

Thoughts?

Reply:
@syrusakbary I actually intended this method to be temporary and I forgot to mention that (just trying to prove this concept). The new and new_with_data could be consolidated into new or we could keep new_with_data if you’d like.

Reply:
I think we can keep both, but one reusing the other so code is not dup.
However I don't have a strong preference, so... your call! :)